### ISSUE TITLE
"i18next::translator: accessing an object - but returnObjects options is not enabled!" in prod build

### ISSUE THREAD

**Issue by stephent** [i18next/i18next#1541]
## 💥 Regression Report

"i18next::translator: accessing an object - but returnObjects options is not enabled!" is reported for certain keys accessed via `t()` and they display with an error message in a production build of a create-react-app app:

"KEY 'PAGE-TITLES.MAP (EN-US)' RETURNED AN OBJECT INSTEAD OF STRING."

instead of 

"Map"

This occurs only for certain keys and only when I set the language to en-US. My en-US translation.json file contains only a subset of keys where the US english version differs from 'en', which is configured as the fallbackLng. The issue is not observed with other languages where every key is translated and there is no expectation of relying on the fallback language 'en'.

It is always the same keys affected, but I can't see any obvious pattern to it.

## Last working version

Worked up to version: 19.8.4
Stopped working in version: 19.8.5

## To Reproduce

With /en/translation.json:

```
{
    "page-titles": {
        "map" : "Map"
    }
}
```

and /en-US/translation.json:

```
{
}
```

(Examples simplified for clarity - en-US/translation.json isn't actually completely empty...)

Switch the language to en-US:

```
i18n.changeLanguage('en-US');
```

```js
console.log(t('page-titles.map')) // returns "KEY 'PAGE-TITLES.MAP (EN-US)' RETURNED AN OBJECT INSTEAD OF STRING."
```

This is observed ONLY in a production build of my create-react-app app. The string displays correctly as 'Map' in my dev environment.

## Expected behavior

In a production build, output when language is set to en-US should be:

```js
console.log(t('page-titles.map')) // should equal 'Map'
```

## Your Environment

- *runtime version*: v14.15.3
- *i18next version*: 19.8.5
- *os*: Mac 
- *any other relevant information*: Create React App, react-scripts 4.0.1

**Comment by jamuhl** [comment]
Guess was introduced with this change: 

https://github.com/i18next/i18next/commit/932f5f662893376254d826e9b01dc4e4c0cd91c0#diff-3274f1a37032fb0ae4e2823def0007c634e869ae0dfc304ff6a12c36513c3a52L44

**Comment by adrai** [comment]
@stephent Can you help to create a reproducable example? https://codesandbox.io/s/react-i18next-forked-520ew?file=/src/app.js

**Comment by stephent** [comment]
@adrai I'll give it a go - although given it only showed up in a prod build, not sure if it will manifest in CSB? If not, I'll try to create a separate CRA repo that can be built.

**Comment by adrai** [comment]
I tried it also based on this example: https://github.com/i18next/react-i18next/tree/master/example/react
and I did nit see that issue 🤷‍♂️

**Comment by stephent** [comment]
I created a repo here trying to reproduce the problem, but no luck so far. It does occur in my main (large-ish) production app, but not here so far, although my i18n config and module versions are identical:

https://github.com/stephent/i18next-issue-1541

If the problem was showing up, I'd expect some of the strings (t("similarities.*") and/or t("page-titles.*") to be displayed as "KEY 'PAGE-TITLES.MAP (EN-US)' RETURNED AN OBJECT INSTEAD OF STRING." or similar on selecting en-US language. I'm testing a build version (i.e. run `build`, then `serve` from package.json). I wonder if it might depend on the hosting environment?

I could potentially push a build of my main app that demos the problem if that would help.

**Comment by adrai** [comment]
Can you try with i18next v19.8.6?

**Comment by stephent** [comment]
Appears to be fixed in 19.8.7 - no longer seeing this in my production build. Thank you!
